File: NavigationBarDemo/NavigationBarTranslucentTest/NavigationBarTranslucentTest/ViewController.m
//
//  ViewController.m
//  NavigationBarTranslucentTest
//
//  Created by 张晨晖 on 2018/5/27.
//  Copyright © 2018年 张晨晖. All rights reserved.
//

#import "ViewController.h"
#import "NavNoneViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    self.title = @"Page One";
    [self setNavVisible];
}

- (void)viewDidDisappear:(BOOL)animated {
    [super viewDidDisappear:animated];
    [self setNavHidden];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.view.backgroundColor = [UIColor greenColor];
}

- (IBAction)pushButtonClick:(UIButton *)sender {
    NavNoneViewController *vc = [[NavNoneViewController alloc] init];
    [self.navigationController pushViewController:vc animated:YES];
}

//显示
- (void)setNavVisible {
    self.navigationController.navigationBar.backgroundColor = nil;
    [self.navigationController.navigationBar setBackgroundImage:nil forBarMetrics:UIBarMetricsDefault];
    [self.navigationController.navigationBar setShadowImage:nil];
    //    [self.navigationController.navigationBar setTintColor:nil];
    //    self.navigationController.navigationBar.translucent = NO;
}

//透明
- (void)setNavHidden {
    self.navigationController.navigationBar.backgroundColor = [UIColor clearColor];
    [self.navigationController.navigationBar setBackgroundImage:[UIImage new] forBarMetrics:UIBarMetricsDefault];
    [self.navigationController.navigationBar setShadowImage:[UIImage new]];
    //    self.navigationController.navigationBar.translucent = YES;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: AppIconChange/APP图标切换/APP图标切换/ViewController.swift
//
//  ViewController.swift
//  APP图标切换
//
//  Created by 张晨晖 on 2018/2/24.
//  Copyright © 2018年 张晨晖. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        ///交换方法去使系统的APP图标已经改变不弹出
        let firstMethod = class_getInstanceMethod(self.classForCoder,#selector(dy_present(_:animated:completion:)))
        let secondMethod = class_getInstanceMethod(self.classForCoder,#selector(present(_:animated:completion:)))
        method_exchangeImplementations(firstMethod!, secondMethod!)
    }

    @objc dynamic func dy_present(_ viewControllerToPresent: UIViewController, animated flag: Bool, completion: (() -> Void)? = nil) {
        if viewControllerToPresent.isKind(of: UIAlertController.classForCoder()) {
            print("yes")
        }
    }

    @IBAction func whiteAppIconBtnClick(_ sender: UIButton) {
        if supportAlternateIcons() == true {
            UIApplication.shared.setAlternateIconName(nil, completionHandler:nil)
        }
    }

    @IBAction func redAppIconBtnClick(_ sender: UIButton) {
        if supportAlternateIcons() == true {
            UIApplication.shared.setAlternateIconName("AppIconRed", completionHandler:nil)
        }
    }

    @IBAction func blueAppIconBtnClick(_ sender: UIButton) {
        if supportAlternateIcons() == true {
            UIApplication.shared.setAlternateIconName("AppIconBlue", completionHandler:nil)
        }
    }

    func supportAlternateIcons() -> Bool {///判断是否支持切换应用的APPIcon
        return UIApplication.shared.supportsAlternateIcons
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

